### PR TITLE
Docs restructure

### PR DESCRIPTION
**Pull request by croth1**
A major restructuring of the documentation :tada:

A preview is available on readthedocs: https://cf-docs-restructure.readthedocs.io/en/docs_restructure/

Things that still need to be done
- [ ] remove requirements_doc.txt

### CODE REVIEW

**Comment by scopatz** [comment]
Thanks @croth1! Looks like there are some conflicts that need resolving.

File: requirements_doc.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+sphinx
+sphinx-rtd-theme
```
Comment (scopatz) [inline requirements_doc.txt]: I *really* dislike the RTD theme. Can you please remove?

File: src/conf.py [hunk 2]
```diff
@@ -101,14 +103,14 @@
...
-todo_include_todos = False
+todo_include_todos = True
 
 
 # -- Options for HTML output ----------------------------------------------
 
 # The theme to use for HTML and HTML Help pages.  See the documentation for
 # a list of builtin themes.
-html_theme = 'alabaster'
+html_theme = 'sphinx_rtd_theme'
```
Comment (scopatz) [inline src/conf.py]: Please revert this line

Comment (croth1) [reply src/conf.py]: I have no strong opinion on the rtd theme, but I really think alabaster does a very poor in making obvious things obvious.

Can we find a compromise that keeps a better user experience but is not RTD?

Comment (scopatz) [reply src/conf.py]: I am not in love with Alabaster, but I do think it is cleaner than RTD. Probably the best thing to do ultimately is to make a theme configuration that matches our landing page (conda-forge.org), but I am definitely not asking you to do that here :)

**Comment by croth1** [comment]
I have now three versions to compare:

- alabaster: https://cf-docs-restructure.readthedocs.io/en/alabaster_version/
- cloud_sp: https://cf-docs-restructure.readthedocs.io/en/cloud_sp/ (Somehow does not show the nav bar hierarchy properly. Can somebody help here?)
- rtd: https://cf-docs-restructure.readthedocs.io/en/docs_restructure/

**Comment by croth1** [comment]
I think we need to discuss how to continue. I feel that intuitive navigation and content structuring is of utmost importance in order to minimise the time users need to spend to find what they are looking for.

IMO, the navigation bar in the alabaster theme has severe short comings in that regard:

- Lack of usage of colors and graphical elements makes it difficult to parse the navigation bar mentally:
   * I find it hard to differentiate between the individual headers, especially if they contain line wraps
   * there is no tracking which section is currently shown
- The maximum depth of the navigation bar seems to be 2. **That is by far not enough to make use of the deeper, more tree-like structure I chose**.

The RTD theme does not seem to have too many friends here (xref: https://github.com/conda-forge/conda-forge.github.io/issues/652).

Following pain points mentioned by @scopatz on gitter: @ocefpaf can you add more to the list?

1. The default color scheme is too dark (without being a dark theme)
2. It doesn't fill the page sufficiently
3. The main div is left-aligned (rather than centered)
4. I am not a fan of the default font
5. I don't the the default code-block backgrounds have a sufficiently high contrast
6. It seems too boxy

1,2,4,5 could probably easily be fixed by adding a few lines of custom css.
3,6 could be more tricky.

@conda-forge/core, to me the feature of having 3-4 levels in the navigation bar seems **very** important for usability. Do you think we can adapt the rtd theme in a way that is a good compromise?

Any suggestions?

**Comment by scopatz** [comment]
Hmm, the cloud navbar seems to have more than 2 levels... https://cloud-sptheme.readthedocs.io/en/latest/index.html

**Comment by scopatz** [comment]
I think it is just that the way you have orgnaized the docs now, no page has a depth greater than 2.  You may be able to point roottarget or logotarget to another option to point it elsewhere https://cloud-sptheme.readthedocs.io/en/latest/cloud_theme.html#structure

**Comment by scopatz** [comment]
Also, if you look at the RTD theme, the navbar doesn't go more than 2 deep anywhere either. That's just how the docs are laid out.

**Comment by croth1** [comment]
> Hmm, the cloud navbar seems to have more than 2 levels... https://cloud-sptheme.readthedocs.io/en/latest/index.html

yea, but somehow it does not work with chapters as a top level (or I haven't figured out how).

It puts the four top level categories (User Documentation, Maintainer Documentation, Organisation Documentation) as pages which are not directly accessible in the nav bar (compare https://cf-docs-restructure.readthedocs.io/en/cloud_sp/ vs. https://cf-docs-restructure.readthedocs.io/en/docs_restructure/)

> I think it is just that the way you have orgnaized the docs now, no page has a depth greater than 2. You may be able to point roottarget or logotarget to another option to point it elsewhere https://cloud-sptheme.readthedocs.io/en/latest/cloud_theme.html#structure

I do have very deep trees from time to time, see e.g.

image

Also, if you look at the RTD theme, the navbar doesn't go more than 2 deep anywhere either. That's just how the docs are laid out.

RDT allows me to configure how many tree levels I would like to have in the nav bar. Here I have the default of 4 (see: https://sphinx-rtd-theme.readthedocs.io/en/stable/configuring.html#confval-navigation_depth)

I have nested even to level 5 some parts of the docs, but I am not sure if going to 5 in the nav bar would not obscure more than help.

**Comment by scopatz** [comment]
> RDT allows me to configure how many tree levels I would like to have in the nav bar. Here I have the default of 4

I think you might need to use https://sphinxcontrib-fulltoc.readthedocs.io/en/latest/ with cloud to get this behaviour.  Cloud itself uses this

**Comment by scopatz** [comment]
Also here is the sphinx conf.py for cloud: https://bitbucket.org/ecollins/cloud_sptheme/src/884f13909ae408909feb92aa6c5256d3b025f5b9/docs/conf.py?at=default&fileviewer=file-view-default

**Comment by croth1** [comment]
@scopatz, I managed to get it to work here: https://cf-docs-restructure.readthedocs.io/en/cloud_sp/

The design looks good to me, what do you think?

**Comment by scopatz** [comment]
Yes, this looks good to me!

**Comment by scopatz** [comment]
I don't mind fiddling with the colors after we get this merged.

File: requirements_doc.txt (hunk 1 above)
Comment (scopatz) [reply requirements_doc.txt]: Also, note that the dependencies are listed/installed here: https://github.com/conda-forge/conda-forge.github.io/blob/master/.ci_scripts/update_docs#L13 This file can probably be removed

File: src/conf.py [hunk 3]
```diff
@@ -54,9 +56,9 @@
 # built documents.
 #
 # The short X.Y version.
-version = u'0.0.1a1'
+version = u'1.0'
```
Comment (scopatz) [inline src/conf.py]: May as well change this to 2019.01, since that is the last big tag we had.

File: requirements_doc.txt (hunk 1 above)
Comment (croth1) [reply requirements_doc.txt]: I added all required packages to the conda command. The requirements file I still need for rendering the preview on RTD. I will remove it right before the PR gets merged.

File: src/conf.py (hunk 3 above)
Comment (croth1) [reply src/conf.py]: Done!

**Comment by scopatz** [comment]
LGTM Thanks!

**Comment by jakirkham** [comment]
Thanks for doing this @croth1. This was a much needed overhaul 😀

**Pull request merged**